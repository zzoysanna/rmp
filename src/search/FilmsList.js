import React, { Component } from 'react';
import FilmPreview from './FilmPreview';
import './styles/film-list.less';

export default class FilmList extends Component {
	render() {
		const {list, pending} = this.props;
		return (
			<>
				{(!!list.length && !pending) && (
					<div className="film-list">
						{list.map((film, i) => {
							return <FilmPreview key={i} film={film}/>;
						})}
					</div>
				)}
				{(!list.length && !pending) && <div className="no-films">No films found</div>}
				{pending && <div className="no-films">Searching...</div>}
			</>
		);
	}
}
