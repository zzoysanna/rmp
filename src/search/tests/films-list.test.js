import React from 'react';
import { shallow } from 'enzyme';
import FilmsList from '../FilmsList';

const film1 = {
	id: 1,
	title: 'test-title-1',
	poster_path: 'test-path-1',
	genres: ['a', 'b'],
	release_date: '1234',
};

const film2 = {
	id: 2,
	title: 'test-title-2',
	poster_path: 'test-path-2',
	genres: ['a', 'b', 'c'],
	release_date: '5678',
};

describe('FilmsList', () => {
	it('should be render list correctly', () => {
		const component = shallow(<FilmsList list={[film1, film2]}/>);
		expect(component).toMatchSnapshot();
	});

	it('should be render empty list correctly', () => {
		const component = shallow(<FilmsList list={[]}/>);
		expect(component).toMatchSnapshot();
	});
});
