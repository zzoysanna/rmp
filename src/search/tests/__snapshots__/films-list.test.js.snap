// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsList should be render empty list correctly 1`] = `ShallowWrapper {}`;

exports[`FilmsList should be render list correctly 1`] = `ShallowWrapper {}`;
